src/App.js

```diff
@@ -0,0 +1,65 @@
+import Form from "./components/Form"
+import {
+  BrowserRouter as Router,
+  Route,
+  Switch
+} from "react-router-dom";
+import "./App.css"
+import Header from "./components/ui/Header/Header"
+import ProductDetails from "./components/ProductDetailsPage/ProductDetails"
+import ProductEdit from "./components/ProductEditPage/ProductEdit"
+import ProductAdd from "./components/ProductAddPage/ProductAdd"
+import { useEffect } from "react";
+import { connect } from 'react-redux'
+import { authSuccess } from "./components/auth/actions"
+import NotFoundPage from "./components/ErrorPages/NotFound"
+import ServerError from "./components/ErrorPages/ServerError"
+import ProductsPage from "./components/ProductsPage/ProductsPage"
+import { useHistory } from "react-router-dom";
+
+function App(props) {
+  let history = useHistory();
+  useEffect(() => {
+    let user = null
+
+    try {
+      user = JSON.parse(localStorage.getItem("visitor"))
+      props.authSuccess(user)
+    } catch (e) {
+      console.log('Failed to get user from localstorage')
+    }
+
+    if (!user) {
+      history.push("/login")
+    } else {
+      history.push("/")
+    }
```

Comment:
This can easily be shortened to

```suggestion
    try {
      const user = JSON.parse(localStorage.getItem("visitor"))
      props.authSuccess(user)
      history.push("/")
    } catch (e) {
      console.log('Failed to get user from localstorage')
      history.push("/login")
    }
```